diff -r 561623f5f81e Eigen/src/Geometry/Quaternion.h
--- a/Eigen/src/Geometry/Quaternion.h	Tue Mar 01 15:02:57 2016 +0100
+++ b/Eigen/src/Geometry/Quaternion.h	Tue May 17 12:56:10 2016 -0700
@@ -276,7 +276,7 @@
   inline Coefficients& coeffs() { return m_coeffs;}
   inline const Coefficients& coeffs() const { return m_coeffs;}
 
-  EIGEN_MAKE_ALIGNED_OPERATOR_NEW_IF(IsAligned)
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW_IF(bool(IsAligned))
 
 protected:
   Coefficients m_coeffs;
